### PR TITLE
Deeparg singularity container fix 2

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

This is a more portable solution to #1397 , by using bindpaths rather than `--fakeroot`, which may not be available on all systems.

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by grst** [comment]
If the file content doesn't matter, could you just bind-mount a file that is available on any POSIX-system, like `/dev/null`?

File: modules/deeparg/downloaddata/main.nf [hunk 1]
```diff
@@ -11,10 +11,11 @@ process DEEPARG_DOWNLOADDATA {
     We have to force singularity to run with --fakeroot to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
     This flag may not be available on all systems and may be considered a security problem. so please document and /or warn for this in your pipeline!
     */
-    containerOptions { "${workflow.containerEngine}" == 'singularity' ? '--fakeroot' : '' }
+    containerOptions { "${workflow.containerEngine}" == 'singularity' ? "-B ${dummy}:/usr/local/lib/python2.7/site-packages/Theano-0.8.2-py2.7.egg-info/PKG-INFO" : '' }
```
Comment (Midnighter) [inline modules/deeparg/downloaddata/main.nf:14]: This is a workaround for Singularity. Nf-core starts Docker with user emulation (not root), does that also cause problems?

**Review by Midnighter — COMMENTED** [review]
I agree with @grst, binding `/dev/null` would be a nicer solution.

File: modules/deeparg/downloaddata/main.nf (hunk 1 above)
Comment (jfy133) [reply modules/deeparg/downloaddata/main.nf:14]: It didn't seem too :shrug:

**Comment by jfy133** [comment]
OK will try `/dev/null`

**Comment by grst** [comment]
`/bin/bash` works :see_no_evil: 

Or maybe `-B $(which bash):/usr/local/lib/python2.7/site-packages/Theano-0.8.2-py2.7.egg-info/PKG-INFO` is slightly safer. 
Bash is, after all, a hard requirement of nextflow, so it should be available everywhere...

**Comment by jfy133** [comment]
`$(which bash)` hasn't failed yet!

**Comment by Midnighter** [comment]
Maybe include a link in the comment to the discussion here on GitHub regarding the borked permissions.

File: modules/deeparg/downloaddata/main.nf [hunk 2]
```diff
@@ -8,10 +8,9 @@ process DEEPARG_DOWNLOADDATA {
         'https://depot.galaxyproject.org/singularity/deeparg:1.0.2--pyhdfd78af_1' :
         'quay.io/biocontainers/deeparg:1.0.2--pyhdfd78af_1' }"
     /*
-    We have to force singularity to run with --fakeroot to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
-    This flag may not be available on all systems and may be considered a security problem. so please document and /or warn for this in your pipeline!
+    We have to force singularity to run with -B to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
```
Comment (jfy133) [inline modules/deeparg/downloaddata/main.nf:11]:
```suggestion
    We have to force singularity to run with -B to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
    Original report: https://github.com/nf-core/funcscan/issues/23
```

File: modules/deeparg/predict/main.nf [hunk 3]
```diff
@@ -9,10 +9,9 @@ process DEEPARG_PREDICT {
         'https://depot.galaxyproject.org/singularity//deeparg:1.0.2--pyhdfd78af_1' :
         'quay.io/biocontainers/deeparg:1.0.2--pyhdfd78af_1' }"
     /*
-    We have to force singularity to run with --fakeroot to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
-    This flag may not be available on all systems and may be considered a security problem. so please document and /or warn for this in your pipeline!
+    We have to force singularity to run with -B to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
```
Comment (jfy133) [inline modules/deeparg/predict/main.nf:12]:
```suggestion
    We have to force singularity to run with -B to allow reading of a problematic file with borked read-write permissions in an upstream dependency (theanos).
        Original report: https://github.com/nf-core/funcscan/issues/23
```

**Comment by jfy133** [comment]
> Maybe include a link in the comment to the discussion here on GitHub regarding the borked permissions.

done

**Pull request merged**